src/parser.rs

```diff
@@ -2890,10 +2903,16 @@ impl<'a> Parser<'a> {
             } else {
                 let if_not_exists =
                     self.parse_keywords(&[Keyword::IF, Keyword::NOT, Keyword::EXISTS]);
-                if self.parse_keyword(Keyword::PARTITION) {
-                    self.expect_token(&Token::LParen)?;
-                    let partitions = self.parse_comma_separated(Parser::parse_expr)?;
-                    self.expect_token(&Token::RParen)?;
+                if self.peek_keyword(Keyword::PARTITION) {
```

Comment:
To be frank I'd rather not. It's logic that shouldn't break the parser for anyone parsing a different dialect and it's abundantly clear that this library is not attempting to be a validation/standards-verification parser. If you have a contrary example where this breaks a parse that should've worked in another context I'll see what I can do to deal with that. I don't think that exists yet though. I lost a lot of time to being confused by the dialect specialization and I'd rather not add more of it.

There isn't any dialect specialization happening anywhere else near this part of the parser that I can recall either.

Reply:
Following are three examples of specialization by dialect inside the code.

My problem is that if you can validate a specific structure for a dialect and know which dialect supports that pattern, you could parse based on the dialect. Since your PR is related to an extension, not a completely different of doing the parse, I don't see why it is a problem to allow Hive and Generic to have that syntax, while other dialects not.

<img width="947" alt="image" src="https://user-images.githubusercontent.com/75763288/194643708-4263175d-257c-489f-8692-78bbb1cfb942.png">
<img width="934" alt="image" src="https://user-images.githubusercontent.com/75763288/194643763-747327c2-a576-44a9-876b-67396e52e1e1.png">
<img width="1163" alt="image" src="https://user-images.githubusercontent.com/75763288/194643813-547c3371-e5c0-4c7b-86d4-3e93aad18b62.png">

Reply:
In general, I think the crate takes the approach of if we can parse the syntax unambiguously we'll accept it even if certain syntax would not be accepted by the various dialects.  This is definitely not totally consistent as @bitemyapp  has noticed

I tend to agree that in this case we aren't adding any ambiguity so this implementation is fine